pkg/edgeview/src/basics.go

```diff
@@ -0,0 +1,608 @@
+}
+
+func checkClientIPMsg(msg string) bool {
+	if strings.HasPrefix(msg, clientIPMsg) {
+		msgs := strings.SplitN(msg, clientIPMsg, 2)
+		if len(msgs) == 2 {
+			addrPort := strings.Split(msgs[1], ":")
+			if net.ParseIP(addrPort[0]) == nil {
+				log.Errorf("received invalid IP %v", msg)
+				return false
+			}
+			if len(addrPort) == 2 {
+				num, err := strconv.Atoi(addrPort[1])
+				if err != nil || num < 1024 || num > 65535 {
+					log.Errorf("received invalid IP %v", msg)
+					return false
+				}
+			} else if len(addrPort) > 2 {
+				log.Errorf("received invalid IP %v", msg)
+				return false
+			}
+			myEvEndPoint = msgs[1]
+			return true
+		}
+	}
+	return false
+}
+
+func evLogger(deb bool) *logrus.Logger {
+	formatter := logrus.JSONFormatter{
+		TimestampFormat: time.RFC3339Nano,
+	}
+	logger := logrus.New()
+	logger.SetFormatter(&formatter)
+	log = base.NewSourceLogObject(logger, agentName, os.Getpid())
+	if deb {
+		logger.SetLevel(logrus.DebugLevel)
+	}
+	return logger
+}
+
+func getBasics() {
+	if runOnServer {
+		if _, err := os.Stat("/config"); err != nil {
+			return
+		}
+	}
+
+	retStr, err := runCmd("ip -4 -br a | grep UP", false, false)
+	if err == nil {
+		lines := splitLines(retStr)
+		ips := []string{}
+		for _, l := range lines {
+			if strings.HasPrefix(l, "bn") {
+				continue
+			}
+			//fmt.Printf("l: %s\n", l)
+			words := strings.Fields(l)
+			n := len(words)
+			ipaddrs := strings.Split(words[n-1], "/")
+			ips = append(ips, ipaddrs[0])
+		}
+		if myEvEndPoint != "" {
+			fmt.Printf("Device IPs: %v; Endpoint IP %s\n", ips, myEvEndPoint)
+		} else {
+			fmt.Printf("Device IPs: %v\n", ips)
+		}
+	}
+	retStr, err = runCmd("cat /persist/status/uuid", false, false)
+	if err == nil {
+		fmt.Printf("  UUID: %s", retStr)
+	}
+	retStr, err = runCmd("cat /config/server", false, false)
+	if err == nil {
+		var printed bool
+		conts := strings.Split(retStr, "zedcloud.")
+		if len(conts) == 2 {
+			cont2s := strings.Split(conts[1], ".zededa.net")
+			if len(cont2s) == 2 {
+				cluster := cont2s[0]
+				controller := strings.Replace(retStr, cluster, "\033[0;93m"+cluster+"\033[0m", 1)
+				fmt.Printf("  Controller: %s", controller)
+				printed = true
+			}
+		}
+		if !printed {
+			fmt.Printf("  Controller: %s", retStr)
+		}
+	}
+	retStr, err = runCmd("cat /run/eve-release && echo -n ', ' && cat /run/eve.id", false, false)
+	if err == nil {
+		fmt.Printf("  EVE-OS release %s", retStr)
+	}
+	fmt.Printf("  Edge-View Ver: %s\n", edgeViewVersion)
+	retStr, err = runCmd("uptime", false, false)
+	if err == nil {
+		ret1 := strings.Split(retStr, " load")
+		uptime := ret1[0]
+		loc, _ := time.LoadLocation("UTC")
+		fmt.Printf("  %v(%v), uptime %s", time.Now().In(loc).Format(time.RFC3339), loc, uptime)
+	}
+	fmt.Println()
+	fmt.Println()
+	closePipe(true)
+}
+
+// dynamically install package for uncommon options
+func addPackage(programName, pkgName string) error {
+	retStr, err := runCmd("which "+programName, false, false)
```

Comment:
Can we not require the callers to know the exact pathname for programName so this check can just be an os.Stat() to check whether e.g., /usr/sbin/tcpdump exists?

Reply:
Ok, let me find out all the programs are and pass in the whole path to it.